src/shared/reducer/product-filter-reducer.js

```diff
@@ -0,0 +1,50 @@
+import * as ACTIONS from '../actions/types';
+import { productFilterState } from '../context/product-filter-state';
+
+export const productFilterReducer = (state, action) => {
+  switch (action.type) {
+    case ACTIONS.FILTER_BY_CATEGORY: {
+      const { categoryName } = action;
+
+      return {
+        ...state,
+        categoryFilter: {
+          ...state.categoryFilter,
+          [categoryName]: !state.categoryFilter[categoryName],
+        },
+      };
+    }
+
+    case ACTIONS.FILTER_BY_RATING: {
```

Comment:
In my view, you don't need to write `ACTIONS.FILTER_BY_NAME` in every reducer, inspite of that `Filter_BY_NAME` makes much more sense and easily readable also.

Reply:
I have created file for all actions types and exporting all actions and importing them as **ACTIONS**.